Solver options

MaximeBouton opened the pull request:
This PR addresses #9 

- `create_policy` no longer exists
- `ValueIterationPolicy` has outer constructors rather than inner ones. 
- the solve method is `solve(solver, mdp)`
- the solver type has new fields: `verbose`, `include_Q` and `init_util`

MaximeBouton committed:
remove create_policy, verbose from solve + outer construtors

MaximeBouton committed:
new option init_util

MaximeBouton committed:
fix bug in constructor, added test

MaximeBouton committed:
test not include Q

rejuvyesh commented on src/vanilla.jl:
```diff
@@ -16,60 +23,51 @@ mutable struct ValueIterationPolicy <: Policy
+
+# constructor with an optinal initial value function argument
+function ValueIterationPolicy(mdp::Union{MDP,POMDP};
+                              utility::Vector{Float64}=Vector{Float64}(0),
```
Why not:

```julia
utility::Vector{Float64}=zeros(n_states(mdp)),
policy::Vector{Int64}=zeros(Int64, n_states(mdp)),
```

Then you won't need the extra if else.

MaximeBouton replied:
Good point, I made the change.

rejuvyesh committed:
untabify

MaximeBouton committed:
simplify constructor

zsunberg reviewed (approved):
Sorry for the delay - I was travelling, but thanks so much for doing this!!! This looks great! Switching to outer constructors makes my heart happy, haha! :heart: :heart:

zsunberg committed:
Deprecated keyword args

I think we should be a little nicer and have a warning for having the keyword arg and not just throw an error.

zsunberg commented:
Just thought of this - test QMDP against this before you merge it!!!

MaximeBouton committed:
test warning

MaximeBouton commented:
I am preparing a PR on QMDP. I will merge this one after

MaximeBouton commented:
JuliaPOMDP/QMDP.jl#10
